documentation/library_guide/kernel_templates/single_pass_scan.rst

```diff
@@ -0,0 +1,203 @@
++===============================================+=====================================================================+
+|  ``q``                                        | SYCL* queue to submit the kernels to.                               |
++-----------------------------------------------+---------------------------------------------------------------------+
+|                                               | The sequences of elements to apply the algorithm to.                |
+|  - ``in_begin``, ``in_end``, ``out_begin`` (1)| Supported sequence types:                                           |
+|  - ``in_rng``, ``out_rng`` (2)                |                                                                     |
+|                                               | - ``sycl::buffer`` (2),                                             |
+|                                               | - :ref:`oneapi::dpl::experimental::ranges::views::all               |
+|                                               |   <viewable-ranges>` (2),                                           |
+|                                               | - :ref:`oneapi::dpl::experimental::ranges::views::subrange          |
+|                                               |   <viewable-ranges>` (2),                                           |
+|                                               | - :ref:`USM pointers <use-usm>` (1),                                |
+|                                               | - :ref:`oneapi::dpl::begin and oneapi::dpl::end                     |
+|                                               |   <use-buffer-wrappers>` (1).                                       |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+|  ``binary_op``                                | A function object that will applied to the elements of the input    |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+|  ``param``                                    | A :doc:`kernel_param <../kernel_configuration>` object.             |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+
+
+**Type Requirements**:
+
+- The element type of sequence to scan must be a C++ integral or floating-point type
+  other than ``bool`` with a width of up to 64 bits.
+- The binary operator must have a known identity. That is, ``sycl::has_known_identity<Op>::value`` must
+  evaluate to true.
```

Comment:
I suppose this is more of a current limitation since we do plan to support other operators in the future. I moved it around and added a link to the table.